packages/ui/src/views/repo/pull-request/pull-request-list-page.tsx

```diff
@@ -39,22 +53,100 @@ const PullRequestListPage: FC<PullRequestPageProps> = ({
+  const { pullRequests, totalPages, page, setPage, openPullReqs, closedPullReqs, setLabelsQuery } =
+    usePullRequestListStore()
+
   const { t } = useTranslationStore()
   const [searchParams] = useSearchParams()
+  const { labels, values: labelValueOptions, isLoading: isLabelsLoading } = useLabelsStore()
 
   const computedPrincipalData = useMemo(() => {
     return principalData || (defaultSelectedAuthor && !principalsSearchQuery ? [defaultSelectedAuthor] : [])
   }, [principalData, defaultSelectedAuthor, principalsSearchQuery])
 
   const [isAllFilterDataPresent, setisAllFilterDataPresent] = useState<boolean>(true)
 
+  const labelsFilterConfig: CustomFilterOptionConfig<keyof PRListFilters, LabelsValue> = {
+    label: t('views:repos.prListFilterOptions.labels.label', 'Label'),
+    value: 'label_by',
+    type: FilterFieldTypes.Custom,
+    parser: {
+      parse: (value: string): LabelsValue => {
+        const result: LabelsValue = {}
+
+        value.split(';').forEach(entry => {
+          const [key, valueStr = ''] = entry.split(':')
+          if (!key) return
+          result[key] = valueStr === 'true' ? true : valueStr
+        })
+
+        return result
+      },
+      serialize: (value: LabelsValue): string => {
+        const parts = Object.entries(value)
+          .map(([key, val]) => {
+            if (!val) return ''
+            return `${key}:${val}`
+          })
+          .filter(Boolean)
+
+        return `${parts.join(';')}`
+      }
+    },
+    filterFieldConfig: {
+      renderCustomComponent: function ({ value, onChange }): ReactNode {
+        return (
+          <LabelsFilter
+            isLabelsLoading={isLabelsLoading}
+            onInputChange={setLabelsQuery}
+            valueOptions={labelValueOptions}
+            labelOptions={labels}
+            onChange={onChange}
+            value={value}
+          />
+        )
+      },
+      renderFilterLabel: (value?: LabelsValue) => {
+        const labelValuesArr = Object.entries(value ?? {}).filter(([_, val]) => val === true || val)
+        const [firstKey, firstValue] = labelValuesArr[0] || []
+        if (!firstValue) return ''
+
+        const labelDetails = labels.find(label => String(label.id) === firstKey)
+        const valueDetails = labelValueOptions[labelDetails?.key ?? '']?.find(value => String(value.id) === firstValue)
+
+        const remainingLabelValues = labelValuesArr.length - 1
+        return (
+          <div className="flex w-max gap-1">
+            {labelDetails && (
+              <LabelMarker
+                key={firstKey}
+                color={labelDetails.color}
+                label={labelDetails.key}
+                value={valueDetails?.value || ''}
+              />
+            )}
+
+            {remainingLabelValues > 0 && (
+              <Badge
+                size="sm"
+                className="text-xs"
```

Comment:
Shall we remove `text-xs` from `Badge`? We can use the `sm` `counter` Badge as it is. Could you try removing the `primary` theme too?

```tsx
<Badge
  size="sm"
  variant="counter"
```